src/server/room-controller.spec.js

```diff
@@ -158,42 +159,139 @@ describe('room-controller', () => {
+    const privateRoom = { roomId: 'roomId', name: 'Mein schöner Raum', owner: 'owner', access: ROOM_ACCESS_LEVEL.private };
+
+    beforeEach(() => {
+      clientDataMapper.mapRoom.resolves(privateRoom);
+
+      roomService.getRoomById.callsFake(roomId => {
+        if (roomId === privateRoom.roomId) {
+          return Promise.resolve(privateRoom);
+        }
+        return Promise.resolve();
+      });
+    });
+
+    describe('when the private room exists', () => {
+      describe('when the room owner calls', () => {
+
+        const request = {
+          params: {
+            roomId: 'roomId'
+          },
+          user: {
+            _id: 'owner'
+          }
+        };
+
+        const invitations = [{ email: 'lae@bucalae.com', sentOn: new Date() }];
+
+        beforeEach(async () => {
+          roomService.getRoomInvitations = sandbox.stub().resolves(invitations);
+          roomService.isRoomMemberOrOwner = sandbox.stub().resolves(true);
+
+          await sut.handleGetRoomPage(request, {});
+        });
+
+        it('should call getRoomById with roomId', () => {
+          sinon.assert.calledWith(roomService.getRoomById, 'roomId');
+        });
+
+        it('should call mapRoom with the room returned by the service', () => {
+          sinon.assert.calledWith(clientDataMapper.mapRoom, privateRoom);
+        });
+
+        it('should call getRoomInvitations', () => {
+          sinon.assert.calledWith(roomService.getRoomInvitations, 'roomId');
+        });
+
+        it('should call pageRenderer with the right parameters', () => {
+          sinon.assert.calledWith(pageRenderer.sendPage, request, {}, PAGE_NAME.room, { roomDetails: privateRoom, invitations });
+        });
+      });
+
+      describe('when a member calls', () => {
+        const request = {
+          params: {
+            roomId: 'roomId'
+          },
+          user: {
+            _id: 'member'
+          }
+        };
 
-    describe('when the room exists', () => {
+        beforeEach(async () => {
+          roomService.getRoomInvitations = sandbox.stub();
+          roomService.isRoomMemberOrOwner = sandbox.stub().resolves(true);
+
+          await sut.handleGetRoomPage(request, {});
+        });
+
+        it('should call getRoomById with roomId', () => {
+          sinon.assert.calledWith(roomService.getRoomById, 'roomId');
+        });
+
+        it('should call mapRoom with the room returned by the service', () => {
+          sinon.assert.calledWith(clientDataMapper.mapRoom, privateRoom);
+        });
+
+        it('should not call getRoomInvitations', () => {
+          sinon.assert.notCalled(roomService.getRoomInvitations);
+        });
+
+        it('should call pageRenderer with the right parameters', () => {
+          sinon.assert.calledWith(pageRenderer.sendPage, request, {}, PAGE_NAME.room, { roomDetails: privateRoom, invitations: null });
+        });
+      });
+
+      describe('when some random person calls', () => {
```

Comment:
`random` could be also a room member. To make it clear I'd say `when the user is not authorized for room access` or similar.

Reply:
Fixed